Fix the CMake build system for distribution packaging:
- require CMake >= 3.10 (CMake 4 rejects < 3.5 compatibility)
- stop forcing CMAKE_BUILD_TYPE, -O3 and -std=c++11; respect user CFLAGS/CXXFLAGS
- build with C++17
- do not write build output to /tmp/build
- do not create symlinks outside DESTDIR; leave udev rules, dll.d and docs to the package
- add BUILD_GUI and BUILD_SCANSDK options (driver-only build without Qt)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b2e0ee2..414b760 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,34 +16,22 @@
 #  
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
+cmake_minimum_required (VERSION 3.10)
 project (epsonscan2)
-set (CMAKE_INSTALL_PREFIX "/usr")
-cmake_minimum_required (VERSION 2.8.12.2)
 
-include(GNUInstallDirs)
-
-if (DEFINED ENV{BUILD_TYPE})
+option(BUILD_GUI "Build the Qt6 graphical front-end (epsonscan2 and detectalert)" ON)
+option(BUILD_SCANSDK "Build the Epson Scan SDK library" ON)
 
-if (${BUILD_TYPE} STREQUAL "Release")
-set(CMAKE_BUILD_TYPE Release)
-elseif (${BUILD_TYPE} STREQUAL "Debug")
-set(CMAKE_BUILD_TYPE Debug)
-endif ()
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
 
-else ()
-set(CMAKE_BUILD_TYPE Release)
-endif()
+include(GNUInstallDirs)
 
-if (DEFINED ENV{EPSON_VERSION})
-else ()
+if (NOT DEFINED EPSON_VERSION)
 set(EPSON_VERSION -1.0.0.0-1)
 endif()
 
-SET(CMAKE_C_FLAGS_DEBUG "-g -O0")
-SET(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++11 -O0")
-SET(CMAKE_C_FLAGS_RELEASE "-O3")
-SET(CMAKE_CXX_FLAGS_RELEASE "-O3 -std=c++11")
-
 ### Boost settings ########
 
 set(Boost_USE_STATIC_LIBS        OFF)
@@ -86,13 +74,9 @@ set(COMMON_ETC_PATH ${CMAKE_INSTALL_FULL_SYSCONFDIR})
 set(EPSON_WORK_PATH /tmp/epsonWork/)
 set(EPSON_SETTINGS_PATH $ENV{HOME}/.epsonscan2/)
 
-SET (CMAKE_INSTALL_PREFIX /usr)
 set(EPSON_INSTALL_PATH ${CMAKE_INSTALL_FULL_LIBDIR}/epsonscan2/)
 set(COMMON_SHARE_PATH ${CMAKE_INSTALL_FULL_DATAROOTDIR})
 
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY /tmp/build)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY /tmp/build)
-
 message("EPSONSCAN_PATH ${EPSONSCAN_PATH}")
 message("COMMON_ETC_PATH ${COMMON_ETC_PATH}")
 message("EPSON_WORK_PATH ${EPSON_WORK_PATH}")
@@ -113,18 +97,5 @@ add_subdirectory(src)
 
 install(DIRECTORY Resources DESTINATION ${EPSON_INSTALL_ROOT}${EPSON_INSTALL_PATH})
 
-install(FILES epsonscan2.rules DESTINATION ${EPSON_INSTALL_ROOT}/lib/udev/rules.d/ RENAME 60-epsonscan2.rules)
-install(FILES epsonscan2 DESTINATION ${EPSON_INSTALL_ROOT}/etc/sane.d/dll.d)
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_FULL_LIBDIR}/sane/)")
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ../epsonscan2/libsane-epsonscan2.so ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_FULL_LIBDIR}/sane/libsane-epsonscan2.so.1)")
-install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ../epsonscan2/libsane-epsonscan2.so ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_FULL_LIBDIR}/sane/libsane-epsonscan2.so.1.0.0)")
-
-install(FILES changelog.Debian DESTINATION ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-
-install(FILES NEWS DESTINATION ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-install(FILES AUTHORS DESTINATION  ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-install(FILES README DESTINATION  ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-install(FILES README_fr DESTINATION  ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-install(FILES COPYING DESTINATION ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-install(FILES LICENSE DESTINATION ${EPSON_INSTALL_ROOT}${CMAKE_INSTALL_PREFIX}/share/doc/epsonscan2${EPSON_VERSION})
-
+# udev rules, the SANE dll.d entry, the libsane symlinks and the
+# documentation are installed by the distribution packaging.
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c884d83..09b1bf6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,7 +16,7 @@
 #  
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.10)
 
 configure_file (${CMAKE_CURRENT_SOURCE_DIR}/Config.h.in
 	  	  	    ${CMAKE_CURRENT_SOURCE_DIR}/Include/Config.h)
@@ -26,7 +26,11 @@ add_subdirectory(ES2Command/Linux)
 add_subdirectory(ES2FileFormat/Linux)
 add_subdirectory(Controller)
 add_subdirectory(SaneWrapper/Linux)
+if (BUILD_GUI)
 add_subdirectory(Standalone)
 add_subdirectory(DetectAlert)
+endif()
+if (BUILD_SCANSDK)
 add_subdirectory(ScanSDK)
+endif()
 
diff --git a/src/DetectAlert/CMakeLists.txt b/src/DetectAlert/CMakeLists.txt
index 22b50ef..dd52ca9 100644
--- a/src/DetectAlert/CMakeLists.txt
+++ b/src/DetectAlert/CMakeLists.txt
@@ -17,7 +17,7 @@
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
 PROJECT(detectalert)
-cmake_minimum_required (VERSION 2.8.10.2)
+cmake_minimum_required (VERSION 3.10)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
diff --git a/src/ES2Command/Linux/CMakeLists.txt b/src/ES2Command/Linux/CMakeLists.txt
index 9e33269..8f3eb84 100644
--- a/src/ES2Command/Linux/CMakeLists.txt
+++ b/src/ES2Command/Linux/CMakeLists.txt
@@ -29,10 +29,7 @@ LINK_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../bin)
 
 # ビルドのタイプはリリース
 #SET(CMAKE_BUILD_TYPE Release)
-SET(CMAKE_C_FLAGS_DEBUG "-g -O0")
-SET(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++11 -Wno-multichar -O0")
-SET(CMAKE_C_FLAGS_RELEASE "-O3")
-SET(CMAKE_CXX_FLAGS_RELEASE "-O3 -std=c++11 -Wno-multichar")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-multichar")
 
 # ビルドするためにインクルードするヘッダーファイルの場所
 INCLUDE_DIRECTORIES(
diff --git a/src/ScanSDK/CMakeLists.txt b/src/ScanSDK/CMakeLists.txt
index 62c95c2..af5f907 100644
--- a/src/ScanSDK/CMakeLists.txt
+++ b/src/ScanSDK/CMakeLists.txt
@@ -17,7 +17,7 @@
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
 PROJECT(epsonscansdk)
-cmake_minimum_required (VERSION 2.8.10.2)
+cmake_minimum_required (VERSION 3.10)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
diff --git a/src/Standalone/CMakeLists.txt b/src/Standalone/CMakeLists.txt
index 13dc55c..002f18f 100644
--- a/src/Standalone/CMakeLists.txt
+++ b/src/Standalone/CMakeLists.txt
@@ -17,7 +17,7 @@
 #  CMakeLists.txt -- template and derived ESC/I ESC/I-2 protocol commands
 
 PROJECT(es2standalone)
-cmake_minimum_required (VERSION 2.8.10.2)
+cmake_minimum_required (VERSION 3.10)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_AUTOMOC ON)
